In [124]:
import pandas as pd
import numpy as np
import scipy.stats as stats

adf = pd.read_csv('alcohol.csv')

This is our database with the alcohol consumption per capita for countries

In [125]:
adf.head()

Entity      Code  Year  \
0     Abkhazia  OWID_ABK  2015   
1  Afghanistan       AFG  2010   
2  Afghanistan       AFG  2015   
3  Afghanistan       AFG  2018   
4  Afghanistan       AFG  2002   

   Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  \
0                                                NaN                                                      
1                                               0.21                                                      
2                                               0.21                                                      
3                                               0.21                                                      
4                                                NaN                                                      

   GDP per capita, PPP (constant 2017 international $)  \
0                                                NaN     
1                                        1957.029070     
2                                        2068.265904     
3                                        2033.804389     
4                                        1189.784668     

   Population (historical estimates) Continent  
0                                NaN      Asia  
1                         29185511.0       NaN  
2                         34413603.0      Asia  
3                         37171922.0       NaN  
4                         22600774.0       NaN

We need to change the column called Entity to Country or region so we can merge it with a df we will create later

In [126]:
adf.rename(columns = {'Entity':'Country or region'}, inplace = True)

We will check that it worked

In [127]:
adf.head()

Country or region      Code  Year  \
0          Abkhazia  OWID_ABK  2015   
1       Afghanistan       AFG  2010   
2       Afghanistan       AFG  2015   
3       Afghanistan       AFG  2018   
4       Afghanistan       AFG  2002   

   Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  \
0                                                NaN                                                      
1                                               0.21                                                      
2                                               0.21                                                      
3                                               0.21                                                      
4                                                NaN                                                      

   GDP per capita, PPP (constant 2017 international $)  \
0                                                NaN     
1                                        1957.029070     
2                                        2068.265904     
3                                        2033.804389     
4                                        1189.784668     

   Population (historical estimates) Continent  
0                                NaN      Asia  
1                         29185511.0       NaN  
2                         34413603.0      Asia  
3                         37171922.0       NaN  
4                         22600774.0       NaN

We only want the collumns for total alcohol consumption, the year, and Country or region so we will make a new dataframe

In [128]:
adf_edit = adf[['Country or region', 'Year', 'Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)']]

In [129]:
adf_edit.head()

Country or region  Year  \
0          Abkhazia  2015   
1       Afghanistan  2010   
2       Afghanistan  2015   
3       Afghanistan  2018   
4       Afghanistan  2002   

   Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  
0                                                NaN                                                     
1                                               0.21                                                     
2                                               0.21                                                     
3                                               0.21                                                     
4                                                NaN

We need to eliminate all rows with NaN as a value so we will make a new df

In [130]:
adf_edit2=adf_edit.dropna()

In [131]:
adf_edit2.head()

Country or region  Year  \
1                    Afghanistan  2010   
2                    Afghanistan  2015   
3                    Afghanistan  2018   
519  Africa Eastern and Southern  2000   
520  Africa Eastern and Southern  2005   

     Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  
1                                             0.210000                                                     
2                                             0.210000                                                     
3                                             0.210000                                                     
519                                           5.014051                                                     
520                                           4.856588

For our sample, we only want data from 2018 so we will again make a new dataframe

In [132]:
adf_2018 = adf_edit2[adf_edit2["Year"] == 2018]

In [133]:
adf_2018.head(10)

Country or region  Year  \
3                     Afghanistan  2018   
523   Africa Eastern and Southern  2018   
554    Africa Western and Central  2018   
586                       Albania  2018   
845                       Algeria  2018   
1218                      Andorra  2018   
1477                       Angola  2018   
1858          Antigua and Barbuda  2018   
2117                   Arab World  2018   
2148                    Argentina  2018   

      Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  
3                                              0.210000                                                     
523                                            5.170911                                                     
554                                            6.835266                                                     
586                                            7.170000                                                     
845                                            0.950000                                                     
1218                                          11.020000                                                     
1477                                           6.940000                                                     
1858                                           6.380000                                                     
2117                                           0.618598                                                     
2148                                           9.650000

Now I want to remove the year from this data frame as we know all the years = 2018 so there's no need to show it in the table

In [134]:
adf2_2018= adf_2018[['Country or region', 'Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)']]

In [135]:
adf2_2018.head(10)

Country or region  \
3                     Afghanistan   
523   Africa Eastern and Southern   
554    Africa Western and Central   
586                       Albania   
845                       Algeria   
1218                      Andorra   
1477                       Angola   
1858          Antigua and Barbuda   
2117                   Arab World   
2148                    Argentina   

      Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  
3                                              0.210000                                                     
523                                            5.170911                                                     
554                                            6.835266                                                     
586                                            7.170000                                                     
845                                            0.950000                                                     
1218                                          11.020000                                                     
1477                                           6.940000                                                     
1858                                           6.380000                                                     
2117                                           0.618598                                                     
2148                                           9.650000

We are going to order the df by country name

In [136]:
order_adf_2018=adf2_2018.sort_values('Country or region')

In [137]:
order_adf_2018.head()

Country or region  \
3                    Afghanistan   
523  Africa Eastern and Southern   
554   Africa Western and Central   
586                      Albania   
845                      Algeria   

     Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  
3                                             0.210000                                                     
523                                           5.170911                                                     
554                                           6.835266                                                     
586                                           7.170000                                                     
845                                           0.950000

We want to calculate the summary statistics for alcohol consumption

In [138]:
order_adf_2018.describe()[['Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)']]

Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)
count                                         235.000000                                                   
mean                                            5.965733                                                   
std                                             3.897767                                                   
min                                             0.003000                                                   
25%                                             2.625000                                                   
50%                                             5.740000                                                   
75%                                             9.065000                                                   
max                                            20.500000

We also want to check for outliers using IQR. Scores falling below the 25% - which is a score of 2.625 - or above the 75% - which is a score of 9.065 - will be considered outliers.

First we will make a new dataframe with all the scores below the 25%

In [139]:
order_adf_below = order_adf_2018[order_adf_2018['Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)'] < 2.625]

We will get the number of scores in this new dataframe to see how any outliers are below the 25%

In [140]:
len(order_adf_below)

59

We are going to display all the outliers below

In [141]:
order_adf_below.head(59)

Country or region  \
3                                            Afghanistan   
845                                              Algeria   
2117                                          Arab World   
4068                                             Bahrain   
4327                                          Bangladesh   
5946                                              Bhutan   
7387                                              Brunei   
9536                            Central African Republic   
9826                                                Chad   
10936                                            Comoros   
13134                       Democratic Republic of Congo   
13652                                           Djibouti   
14808                                              Egypt   
15585                                            Eritrea   
16363                                           Ethiopia   
20412                                          Guatemala   
20672                                             Guinea   
23222                                          Indonesia   
23481                                               Iran   
23740                                               Iraq   
25359                                             Jordan   
26136                                           Kiribati   
26379                                             Kuwait   
27570                                            Lebanon   
28347                                              Libya   
29522                                         Madagascar   
30040                                           Malaysia   
30299                                           Maldives   
30521                                               Mali   
31373                                         Mauritania   
32222                               Micronesia (country)   
32444                         Middle East & North Africa   
32475            Middle East & North Africa (IDA & IBRD)   
32506  Middle East & North Africa (excluding high inc...   
33639                                            Morocco   
33898                                         Mozambique   
34898                                              Nepal   
36096                                              Niger   
38086                                               Oman   
38366                        Pacific island small states   
38397                                           Pakistan   
39359                                   Papua New Guinea   
41094                                              Qatar   
44032                                       Saudi Arabia   
44292                                            Senegal   
45293                                          Singapore   
46159                                    Solomon Islands   
46412                                            Somalia   
48343                                              Sudan   
49435                                              Syria   
50730                                              Timor   
50989                                               Togo   
51320                                              Tonga   
51811                                            Tunisia   
52070                                             Turkey   
52692                                             Tuvalu   
54612                                         Uzbekistan   
54871                                            Vanuatu   
56306                                              Yemen   

       Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  
3                                               0.210000                                                     
845                                             0.950000                                                     
2117                                            0.618598                                                    

Now we will make a new dataframe with all the scores above the 75%

In [142]:
order_adf_above = order_adf_2018[order_adf_2018['Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)'] > 9.065]

We will get the number of scores in this new dataframe to see how any outliers are above the 75%

In [143]:
len(order_adf_above)

59

We are going to display all the outliers below

In [144]:
order_adf_above.head(59)

Country or region  \
1218                          Andorra   
2148                        Argentina   
3031                        Australia   
3290                          Austria   
4586                         Barbados   
4838                          Belarus   
5097                          Belgium   
7646                         Bulgaria   
7905                     Burkina Faso   
9795   Central Europe and the Baltics   
10157                           Chile   
11176                           Congo   
12025                         Croatia   
12615                          Cyprus   
12874                         Czechia   
13393                         Denmark   
13911                        Dominica   
15845                         Estonia   
16104                        Eswatini   
16622                       Euro area   
16912           Europe & Central Asia   
17005                  European Union   
17509                         Finland   
17773                          France   
19004                         Germany   
19594                          Greece   
19957                         Grenada   
21741                     High income   
22290                         Hungary   
22704                         Iceland   
23999                         Ireland   
26897                            Laos   
27280                          Latvia   
28865                       Lithuania   
29191                      Luxembourg   
32569                         Moldova   
33307                      Montenegro   
35157                     Netherlands   
35600                     New Zealand   
36355                         Nigeria   
36687                   North America   
37796                    OECD members   
40397                          Poland   
40656                        Portugal   
40915       Post-demographic dividend   
41453                         Romania   
41712                          Russia   
42620                     Saint Lucia   
44812                      Seychelles   
45615                        Slovakia   
45874                        Slovenia   
46672                    South Africa   
47253                     South Korea   
47735                           Spain   
49176                     Switzerland   
50212                        Tanzania   
52916                          Uganda   
53693                  United Kingdom   
53953                   United States   

       Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  
1218                                           11.020000                                                     
2148                                            9.650000                                                     
3031                                           10.510000                                                     
3290                                           11.960000                                                     
4586                                            9.690000                                                     
4838                                           11.450000                                                     
5097                                           11.080000                                                     
7646                                           12.650000                                                     
7905                                           12.030000                                                     
9795                                           11.932080                                                     
10157                                           9.070000                                                     
11176                                           9.270000                                                     
12025                                           9.230000                                                     
12615              

We now have two dataframes displaying all the outliers for alcohol consumption

Next we need to load our second df for 2018 happiness scores

In [145]:
hdf = pd.read_csv('happy_2018.csv')

In [146]:
hdf.head()

Overall rank Country or region  Score  GDP per capita  Social support  \
0             1           Finland  7.632           1.305           1.592   
1             2            Norway  7.594           1.456           1.582   
2             3           Denmark  7.555           1.351           1.590   
3             4           Iceland  7.495           1.343           1.644   
4             5       Switzerland  7.487           1.420           1.549   

   Healthy life expectancy  Freedom to make life choices  Generosity  \
0                    0.874                         0.681       0.202   
1                    0.861                         0.686       0.286   
2                    0.868                         0.683       0.284   
3                    0.914                         0.677       0.353   
4                    0.927                         0.660       0.256   

   Perceptions of corruption  
0                      0.393  
1                      0.340  
2                      0.408  
3                      0.138  
4                      0.357

We only want the columns for the country and the score so we will make a new df

In [147]:
hdf2= hdf[['Country or region', 'Score']]

In [148]:
hdf2.head()

Country or region  Score
0           Finland  7.632
1            Norway  7.594
2           Denmark  7.555
3           Iceland  7.495
4       Switzerland  7.487

We are going to order the df by country name

In [149]:
order_hdf=hdf2.sort_values('Country or region')

In [150]:
order_hdf.head(10)

Country or region  Score
144       Afghanistan  3.632
111           Albania  4.586
83            Algeria  5.295
141            Angola  3.795
28          Argentina  6.388
128           Armenia  4.321
9           Australia  7.272
11            Austria  7.139
86         Azerbaijan  5.201
42            Bahrain  6.105

We want to calculate the summary statistics for average happyness score

In [151]:
order_hdf.describe()[['Score']]

Score
count  156.000000
mean     5.375917
std      1.119506
min      2.905000
25%      4.453750
50%      5.378000
75%      6.168500
max      7.632000

We also want to check for outliers using IQR. Scores falling below the 25% - which is a score of 4.45375 - or above the 75% - which is a score of 6.1685 - will be considered outliers.

First we will make a new dataframe with all the scores below the 25%

In [152]:
order_hdf_below = order_hdf[order_hdf["Score"] < 4.45375]

We will get the number of scores in this new dataframe to see how any outliers are below the 25%

In [153]:
len(order_hdf_below)

39

We are going to display all the outliers below

In [154]:
order_hdf_above.head(39)

Country or region  Score
28             Argentina  6.388
9              Australia  7.272
11               Austria  7.139
15               Belgium  6.927
27                Brazil  6.419
6                 Canada  7.328
24                 Chile  6.476
36              Colombia  6.260
12            Costa Rica  7.072
20        Czech Republic  6.711
2                Denmark  7.555
0                Finland  7.632
22                France  6.489
14               Germany  6.965
29             Guatemala  6.382
3                Iceland  7.495
13               Ireland  6.977
18                Israel  6.814
16            Luxembourg  6.910
34              Malaysia  6.322
21                 Malta  6.627
23                Mexico  6.488
5            Netherlands  7.441
7            New Zealand  7.324
1                 Norway  7.594
26                Panama  6.430
31                 Qatar  6.374
32          Saudi Arabia  6.371
33             Singapore  6.343
38              Slovakia  6.173
35                 Spain  6.310
8                 Sweden  7.314
4            Switzerland  7.487
25                Taiwan  6.441
37     Trinidad & Tobago  6.192
19  United Arab Emirates  6.774
10        United Kingdom  7.190
17         United States  6.886
30               Uruguay  6.379

Now we will make a new dataframe with all the scores above the 75%

In [155]:
order_hdf_above = order_hdf[order_hdf["Score"] > 6.1685]

We will get the number of scores in this new dataframe to see how any outliers are above the 75%

In [156]:
len(order_hdf_above)

39

We are going to display all the outliers below

In [157]:
order_hdf_above.head(39)

Country or region  Score
28             Argentina  6.388
9              Australia  7.272
11               Austria  7.139
15               Belgium  6.927
27                Brazil  6.419
6                 Canada  7.328
24                 Chile  6.476
36              Colombia  6.260
12            Costa Rica  7.072
20        Czech Republic  6.711
2                Denmark  7.555
0                Finland  7.632
22                France  6.489
14               Germany  6.965
29             Guatemala  6.382
3                Iceland  7.495
13               Ireland  6.977
18                Israel  6.814
16            Luxembourg  6.910
34              Malaysia  6.322
21                 Malta  6.627
23                Mexico  6.488
5            Netherlands  7.441
7            New Zealand  7.324
1                 Norway  7.594
26                Panama  6.430
31                 Qatar  6.374
32          Saudi Arabia  6.371
33             Singapore  6.343
38              Slovakia  6.173
35                 Spain  6.310
8                 Sweden  7.314
4            Switzerland  7.487
25                Taiwan  6.441
37     Trinidad & Tobago  6.192
19  United Arab Emirates  6.774
10        United Kingdom  7.190
17         United States  6.886
30               Uruguay  6.379

Now we have two dataframes displaying all the outliers for happiness scores

Next we will go back to the happiness and alcohol consumption dataframes and merge them into one

In [158]:
comb_df=pd.merge(order_adf_2018, order_hdf, how='outer', on='Country or region')

In [159]:
comb_df.head()

Country or region  \
0                  Afghanistan   
1  Africa Eastern and Southern   
2   Africa Western and Central   
3                      Albania   
4                      Algeria   

   Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  \
0                                           0.210000                                                      
1                                           5.170911                                                      
2                                           6.835266                                                      
3                                           7.170000                                                      
4                                           0.950000                                                      

   Score  
0  3.632  
1    NaN  
2    NaN  
3  4.586  
4  5.295

Now that they are merged, we need to eliminate any rows without a score and make a new df

In [160]:
final_comb_df= comb_df.dropna()

In [161]:
final_comb_df.head()

Country or region  \
0       Afghanistan   
3           Albania   
4           Algeria   
6            Angola   
9         Argentina   

   Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  \
0                                               0.21                                                      
3                                               7.17                                                      
4                                               0.95                                                      
6                                               6.94                                                      
9                                               9.65                                                      

   Score  
0  3.632  
3  4.586  
4  5.295  
6  3.795  
9  6.388

We want to answer the following question: 
Do countries with a happiness score greater than 6 have an average total alcohol consumption per capita less than  5.96?
To answer this we will want to run a left-sided, single sample t-test.
Our hypotheses would be:
Null >= 5.96
Alternative μ < 5.96
We will be using the final_comb_df data frame to run the test.

In [162]:
final_comb_df.head()

Country or region  \
0       Afghanistan   
3           Albania   
4           Algeria   
6            Angola   
9         Argentina   

   Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  \
0                                               0.21                                                      
3                                               7.17                                                      
4                                               0.95                                                      
6                                               6.94                                                      
9                                               9.65                                                      

   Score  
0  3.632  
3  4.586  
4  5.295  
6  3.795  
9  6.388

In [163]:
df_above_6 = final_comb_df[final_comb_df["Score"] > 6]

In [164]:
df_above_6.head()

Country or region  \
9          Argentina   
11         Australia   
12           Austria   
15           Bahrain   
19           Belgium   

    Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  \
9                                                9.65                                                      
11                                              10.51                                                      
12                                              11.96                                                      
15                                               1.09                                                      
19                                              11.08                                                      

    Score  
9   6.388  
11  7.272  
12  7.139  
15  6.105  
19  6.927

Now we have our sample of scores above 6 from 2018. To run the t-test the only data from the sample we need is the total alcohol consumption. So we will make a new dataframe with only that data

In [165]:
test2_df = df_above_6[['Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)']]

In [166]:
test2_df.head(10)

Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)
9                                                9.65                                                   
11                                              10.51                                                   
12                                              11.96                                                   
15                                               1.09                                                   
19                                              11.08                                                   
26                                               7.42                                                   
33                                               8.94                                                   
39                                               9.07                                                   
41                                               5.74                                                   
44                                               4.87

Now we can run the t-test and get the t-stat. We will be using a critical level/alpha of .05 as that's what's most common to use

In [167]:
stats.ttest_1samp(a=test2_df, popmean = 5.96, alternative='less')

Ttest_1sampResult(statistic=array([2.81467281]), pvalue=array([0.99629947]))

The t test statistic is 2.81467281

We need to get the t-critical value and compare it to the t-stat inorder to determine if we accept or reject the null. to get this we first need to determine the degrees of freedom

In [168]:
dof=len(test2_df.index)-1

In [169]:
print(dof)

42


So our degrees of freedom is 42

Now we use our chosen alpha level and the degrees of freedom to find the t-critical value

In [170]:
stats.t.ppf(.05,42)

-1.6819523559426006

We can see our t-critical value is -1.6819523559426006

Our t test statistic of 2.811 is less extreme than our t-critical value of -1.68. This tells us we should accept the null hypothesis. Additionally our p-value of 0.99999262 is greater than our alpha of .05 which also tells us we should accept the null. So we can concluded that countries with an average happiness score greater than 6 have an average total alcohol consumption per capita that is greater than or equal to 5.96.



We want to answer the following question: Do countries with a happiness score less than 5 have an average total alcohol consumption per capita greater than 5.96? To answer this we will want to run a right-sided, single sample t-test. Our hypotheses would be: Null <= 5.96 Alternative μ > 5.96

So first we need to make a dataframe with countries with happiness scores below 3.

In [171]:
df_below_5 = final_comb_df[final_comb_df["Score"] < 5]

In [172]:
df_below_5.head()

Country or region  \
0        Afghanistan   
3            Albania   
6             Angola   
10           Armenia   
16        Bangladesh   

    Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)  \
0                                               0.210                                                      
3                                               7.170                                                      
6                                               6.940                                                      
10                                              5.550                                                      
16                                              0.019                                                      

    Score  
0   3.632  
3   4.586  
6   3.795  
10  4.321  
16  4.500

Now we have our sample of scores below 5 from 2018. To run the t-test the only data from the sample we need is the total alcohol consumption. So we will make a new dataframe with only that data

In [173]:
test3_df = df_below_5[['Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)']]

In [174]:
test3_df.head()

Total alcohol consumption per capita (liters of pure alcohol, projected estimates, 15+ years of age)
0                                               0.210                                                   
3                                               7.170                                                   
6                                               6.940                                                   
10                                              5.550                                                   
16                                              0.019

Now we can run the t-test and get the t-stat. We will be using a critical level/alpha of .05 as that's what's most common to use

In [175]:
stats.ttest_1samp(a=test3_df, popmean = 5.96, alternative='greater')

Ttest_1sampResult(statistic=array([-2.91869939]), pvalue=array([0.99742565]))

The t test statistic is -2.91869939

We need to get the t-critical value and compare it to the t-stat inorder to determine if we accept or reject the null. to get this we first need to determine the degrees of freedom

In [176]:
dof=len(test3_df.index)-1

In [177]:
print(dof)

53


So our degrees of freedom is 53

Now we use our chosen alpha level and the degrees of freedom to find the t-critical value

In [178]:
stats.t.ppf(1-.05,53)

1.6741162362294313

Our t test statistic of -2.92 is less extreme than our t-critical value of 1.67. This tells us we accept reject the null hypothesis. Additionally our p-value of 0.99742565 is greater than our alpha of .05 which also tells us we should accept the null. So we can concluded that countries with an average happiness score less than 5 have an average total alcohol consumption per capita that is greater than or equal to 5.96.